msgpack-core/src/main/java/org/msgpack/core/buffer/MessageBuffer.java

```diff
@@ -225,13 +263,43 @@ public static void releaseBuffer(MessageBuffer buffer)
         this.base = arr;
         this.address = ARRAY_BYTE_BASE_OFFSET + offset;
         this.size = length;
+        this.reference = null;
+    }
+
+    /**
+     * Create a MessageBuffer instance from a given ByteBuffer instance
+     *
+     * @param bb
+     */
+    MessageBuffer(ByteBuffer bb)
+    {
+        if (bb.isDirect()) {
+            if (isUniversalBuffer) {
+                throw new IllegalStateException("Cannot create MessageBuffer from DirectBuffer");
```

Comment:
let's include "on this platform" in the message so that people don't think it's not MessagePack's fault...

```
+                throw new IllegalStateException("Cannot create MessageBuffer from DirectBuffer on this platform");
```